external_samples/expansion_hub_motor.py

```diff
@@ -0,0 +1,126 @@
+
+This component wraps the expansion_hub.ExpansionHubMotor class, providing
+support for a motor connected to a REV Expansion Hub.
+"""
+
+__author__ = "lizlooney@google.com (Liz Looney)"
+
+from typing import Self
+from component import Component, PortType, InvalidPortException
+import expansion_hub
+import wpimath
+
+# TODO(lizlooney): Update port types.
+
+class ExpansionHubMotor(Component):
+    def __init__(self, ports : list[tuple[PortType, int]]):
+        port_type, hub_number = ports[0]
+        if port_type != PortType.USB_PORT:
+            raise InvalidPortException
+        port_type, motor_number = ports[1]
+        if port_type != PortType.USB_PORT:
+            raise InvalidPortException
+        self.expansion_hub_motor = expansion_hub.ExpansionHubMotor(hub_number, motor_number)
+
+    def get_manufacturer(self) -> str:
+        return "REV Robotics"
+
+    def get_name(self) -> str:
+        return "Expansion Hub Motor"
+
+    def get_part_number(self) -> str:
+        return ""
+
+    def get_url(self) -> str:
+        return ""
+
+    def get_version(self) -> tuple[int, int, int]:
+        return (1, 0, 0)
+
+    def start(self) -> None:
+        self.expansion_hub_motor.setEnabled(True)
+
+    def update(self) -> None:
+        pass
+
+    def stop(self) -> None:
+        # TODO: Send stop command to motor.
+        pass
+
+    def reset(self) -> None:
+        pass
+
+    def get_connection_port_type(self) -> list[PortType]:
+        return [PortType.USB_PORT, PortType.USB_PORT]
+
+    def periodic(self) -> None:
+        pass
+
+    # Alternative constructor to create an instance from a hub number and a motor port.
+    @classmethod
+    def from_hub_number_and_motor_number(cls: type[Self], hub_number: int, motor_number: int) -> Self:
+        return cls([(PortType.USB_PORT, hub_number), (PortType.USB_PORT, motor_number)])
+
+    # Component specific methods
+
+    # Methods from expansion_hub.ExpansionHubMotor
+
+    def setPercentagePower(self, power: float):
+        self.expansion_hub_motor.setPercentagePower(power)
+
+    def setVoltage(self, voltage: wpimath.units.volts):
+        self.expansion_hub_motor.setVoltage(voltage)
+
+    def setPositionSetpoint(self, setpoint: float):
+        self.expansion_hub_motor.setPositionSetpoint(setpoint)
+
+    def setVelocitySetpoint(self, setpoint: float):
+        self.expansion_hub_motor.setVelocitySetpoint(setpoint)
+
+    def setEnabled(self, enabled: bool):
+        self.expansion_hub_motor.setEnabled(enabled)
+
+    def setFloatOn0(self, floatOn0: bool):
+        self.expansion_hub_motor.setFloatOn0(floatOn0)
+
+    def getCurrent(self) -> float:
+        return self.expansion_hub_motor.getCurrent()
+
+    def setDistancePerCount(self, perCount: float):
+        self.expansion_hub_motor.setDistancePerCount(perCount)
+
+    def isHubConnected(self) -> bool:
+        return self.expansion_hub_motor.isHubConnected()
+
+    def getEncoder(self) -> float:
```

Comment:
I thought encoder ticks were int instead of float

Reply:
It's float in the class from Thad. See https://github.com/wpilibsuite/systemcore-blocks-interface/blob/485a6a70d054ed7cc26c6a9231ac0afee5852cd4/server_python_scripts/expansion_hub.py#L162